### ISSUE TITLE
Customize Codelens Font

### ISSUE THREAD

**Issue by GitClickOk** [microsoft/vscode#16038]
- VSCode Version:
1.7.2
- OS Version:
Windows 10

Steps to Reproduce:

The actual implementation of Codelens on VSCode is annoying. I did a research, and I can disable Codelens with "editor.codeLens": false. 

But I think will be good if, instead of full disable, I can customize the font size. If the codelens have at least on half of the current size, maybe it can be usable... With the current size, it looks with an actual line of code. So, maybe you can do some options like:
"editor.codeLens.fontFamily": "Consolas, 'Courier New', monospace",
"editor.codeLens.fontWeight": "normal",
"editor.codeLens.fontSize": 7,

**Comment by nodrogyad** [comment]
This would be a great addition as codelens currently is almost unusable with a "editor.fontSize" < 14

**Comment by sillyotter** [comment]
In addition to setting the size, weight, etc, I also noticed while trying to track this down this morning that the vertical-alignment for the thing is set to sub, which clips the bottom of the line off if your not careful. Even if we can change the fonts, this might need to be set to baseline to prevent clipping.

**Comment by Xenya0815** [comment]
What's status? It would be really good to be able to change the font size.

**Comment by feafarot** [comment]
This feature would be really handy. Usually, you want to decrease the size of "codelens" text since this information is minor and should not take that much space.

**Comment by jrieken** [comment]
How about a PR? Anyone? I am happy to advice.

**Comment by boyedarat** [comment]
I would like CondeLens to pick up the fontFamily and fontLigatures from the editor:
`this._domNode.style.fontFamily=fontInfo.fontFamily
this._domNode.style.fontLigatures=fontInfo.fontLigatures`

Would it be possible to do the above at least for now?

**Comment by TomasHubelbauer** [comment]
@jrieken I'd like to try implementing the support for setting the CodeLens font size. Would you be able to advise on where to start and where to look trying to implement this?

**Comment by jrieken** [comment]
Sure, `codeLensWidget.ts` is your starting point. Our code lens is slight weird as there are two pieces involved: (1) a [view zone](https://github.com/Microsoft/vscode/blob/master/src/vs/editor/contrib/codelens/codelensWidget.ts#L24) which makes space between lines and (2) a [content widget](https://github.com/Microsoft/vscode/blob/master/src/vs/editor/contrib/codelens/codelensWidget.ts#L54) which we position onto that space. So, the former defines the height (today as being that of one line, you can also have `hightInPx` tho) and the latter is the actual rendering and the place where different fonts styles go. Just checkout that code, esp [this](https://github.com/Microsoft/vscode/blob/master/src/vs/editor/contrib/codelens/codelensWidget.ts#L110). Then add a new editor configuration which is a bit boilerplate: check out `editorOptions.ts` and follow all occurrences of an existing option, e.g `showFoldingControls:boolean` or `codeLens:boolean`.

**Comment by boyedarat** [comment]
@TomasHubelbauer Any luck with a fix for this?

**Comment by TomasHubelbauer** [comment]
@boyedarat I haven't picked it up yet, will at some point when I have time I want to dedicate to OSS. Feel free to pick this up in parallel if you want to, I cannot commit to a specific deadline if you're waiting on this.

**Comment by boyedarat** [comment]
@jrieken any plans to merge this?

**Comment by mikeerickson** [comment]
I too would love to see this. This appearance (assuming what you are speaking of) is quite intrusive for my liking.  Is this information already stylable?

![image](https://user-images.githubusercontent.com/183153/51721448-b0d2ac00-2006-11e9-9e47-706b740edee5.png)

**Comment by cr0cK** [comment]
Up

**Comment by Feofilakt** [comment]
Correct me if I'm wrong, the MR with half-done work has been merged into repository?

**Comment by TomasHubelbauer** [comment]
That's correct. I haven't heard from the VS Code devs to know if I'm going about this the right way so I lost interest in working in this further. Figuring out the internals of VS Code takes some time and I didn't want to spend it not knowing if my PR will be rejected because I implemented it in an incorrect way or ignored altogether.

**Comment by TomasHubelbauer** [comment]
It has not been merged, I have closed it as a part of spring cleaning after it has gotten no feedback for a year.

**Comment by Feofilakt** [comment]
Oh, I see. Thank you for attempt. It is sorry that misrosoft's team not consider this feature as priority.

**Comment by TomasHubelbauer** [comment]
I might pick it up again when I'm willing to invest more time into VS Code specifically. I sympathize with the VS Code members being busy and unable to guide each and every person attempting to contribute. Maybe one day the VS Code team will have a dedicated community person whose job will be supporting would-be contributors rather than coding themselves. That'd be cool 🙂

**Comment by Feofilakt** [comment]
They could be done this themselves. Obviosly they don't consider this feature much priority for some reason.

**Comment by dannymcgee** [comment]
I'm taking a look at this now — can't make any promises, but I've just discovered CodeLens for TypeScript and I love it but desperately want to make it smaller, so motivation is high. 😉 The code looks pretty straightforward, but I'll reach out if I run into any roadblocks.

**Comment by mikeerickson** [comment]
@dannymcgee while in there, i would love to set the background color as well to at least make it easier to reason about, difficult to use as-is on a 13” screen using Material Theme

**Comment by dannymcgee** [comment]
> @dannymcgee while in there, i would love to set the background color as well to at least make it easier to reason about, difficult to use as-is on a 13” screen using Material Theme

That might be a little trickier (and maybe tougher to get approved) since it'll require messing with the theme API, but I'll start with the font/font size/line height and see where I can go from there.

**Comment by jrieken** [comment]
Next insiders will have these settings: `editor.codeLensFontSize` and `editor.codeLensFontFamily`

<img width="876" alt="Screenshot 2020-11-09 at 21 58 22" src="https://user-images.githubusercontent.com/1794099/98595789-bb8f2b80-22d6-11eb-81f7-76fdaa1ccb6a.png">

**Comment by jrieken** [comment]
To _verify_

* make sure that there are two code lens font settings (size, face) that allow to customise the code lense appearance
* try font features like ligatures
